internal/pkg/probe/probe.go

```diff
@@ -65,3 +67,35 @@ func (m *deviceScanner) ScanDevices(ctx context.Context, log logr.Logger) ([]str
 	}
 	return paths, nil
 }
+
+// GetUnfomattedDevices retrieves devices that are unformatted.
+func (m *deviceScanner) ScanAvailableDevices(ctx context.Context, log logr.Logger) (*block.DeviceList, error) {
+	devices, err := m.GetDevices(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get devices: %w", err)
+	}
+
+	var unformatted []block.Device
+	for _, device := range devices.Devices {
+		if !m.filter.Match(device) {
+			log.V(2).Info("device filtered out", "device", device)
+			continue
+		}
+		isUnformatted, err := m.IsBlkDevUnformatted(device.Path)
+		if err != nil {
+			log.V(2).Error(err, "failed to check if device is unformatted", "device", device)
```

Comment:
The caller logs the error so I'd suggest removing the log here to avoid duplicates.

Reply:
I think it's still logging the error twice?

Reply:
Yes, you are right - missed it!